libs/constrained-generators/src/Constrained/Base.hs

```diff
@@ -1331,8 +1358,9 @@ backPropagation (SolverPlan plan graph) = SolverPlan (go [] (reverse plan)) grap
 
         termVarEqCases :: HasSpec fn b => Specification fn a -> Var b -> Term fn b -> [SolverStage fn]
         termVarEqCases (MemberSpec vs) x' t
-          | Set.singleton (Name x) == freeVarSet t =
-              [SolverStage x' [] $ MemberSpec $ nub [errorGE $ runTerm (singletonEnv x v) t | v <- vs]]
+          | Set.singleton (Name x) == freeVarSet t
+          , Just ys <- NE.nonEmpty (nub [errorGE $ runTerm (singletonEnv x v) t | v <- NE.toList vs]) =
+              [SolverStage x' [] $ MemberSpec ys]
```

Comment:
This is not the same semantics as before. It should be `ErrorSpec` if the list is empty. Use the utility function you defined above.

Reply:
Now it is
```
 termVarEqCases (MemberSpec vs) x' t
          | Set.singleton (Name x) == freeVarSet t =
              [SolverStage x' [] $ memberSpecList (nub [errorGE $ runTerm (singletonEnv x v) t | v <- vs])
                                                  (NE.toList ["None of the values"
                                                             ," "++show vs
                                                             ,"backPropagate successfully"]) ] 
```
Not sure the error message makes sense.